src/query/builder.js

```diff
@@ -68,6 +68,44 @@ assign(Builder.prototype, {
     return this;
   },
 
+  // With
+  // ------
+
+  with(alias, statement, bindings) {
+    if (typeof statement === 'function') {
+      return this.withWrapped(alias, statement);
+    }
+    // Allow a raw statement to be passed along to the query.
+    if (statement instanceof Raw && arguments.length >= 2) {
+      return this.withRaw(alias, statement, bindings);
+    }
+
```

Comment:
are you talking about the `return` in line 83 ?
i could replace `return` by `return this` like other functions : 
there is no other function who throw any exception in this file...
would you like me to `return` by:
`throw new Error("invalid arguments for with()");`